### PR TITLE
Track last state check and use it to adjust state if needed

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

**Comment by Dieterbe** [comment]
my biggest concern here is the 3rd commit. in theory i'm not supposed to modify an existing migration but from what torkel said it should be fine. also moving the new one in front of the older one seems unusual but needed.

**Comment by Dieterbe** [comment]
@woodsaj thoughts?

**Comment by Dieterbe** [comment]
I just confirmed that modifying the existing migration is harmless in this case.
```
 79 func NewAddColumnMigration(table Table, col *Column) *AddColumnMigration {
 80   return &AddColumnMigration{tableName: table.Name, column: col}
 81 }
```
```
grep -A1 'var monitorV. = Table' pkg/services/sqlstore/migrations/monitor_mig.go
	var monitorV3 = Table{
		Name: "monitor",
--
	var monitorV4 = Table{
		Name: "monitor",
```

the reordering is still interesting though

File: pkg/services/sqlstore/monitor.go [hunk 1]
```diff
@@ -44,6 +45,20 @@ type MonitorWithCollectorDTO struct {
...
 }
 
+// scrutinizeState fixes the state.  We can't just trust what the database says, we have to verify that the value actually has been updated recently.
+// this could later be made more precise by taking into account the current scheduler offset, the range it is assumed to be in (say 0 to 50 seconds)
+// and the offset of the monitor.  But for now, we simply allow 1 run and an additional minute of delay.
+func scrutinizeState(state m.CheckEvalResult, stateCheck time.Time, frequency int64) m.CheckEvalResult {
+	if state == m.EvalResultUnknown {
+		return state
+	}
+	maxDelay := time.Duration(frequency+60) * time.Second
```
Comment (woodsaj) [inline pkg/services/sqlstore/monitor.go]: I personally would prefer to have the maxDelay be a multiple of the frequency.  ie, the state should be unknown if 2 consecutive checks failed to run.

**Comment by Dieterbe** [comment]
pushed a commit with your suggested change. how does it look now @woodsaj ?

**Comment by woodsaj** [comment]
looks good to me.

**Pull request merged**